python/common/org/python/types/Complex.java

```diff
@@ -162,31 +162,11 @@ private String partToStr(org.python.types.Float x) {
             __doc__ = "Return repr(self)."
     )
     public org.python.Object __repr__() {
-        java.lang.StringBuilder buffer = new java.lang.StringBuilder();
-        boolean real_present = true;
-        if (this.real.value != 0) {
-            buffer.append("(");
-            if (((org.python.types.Bool) ((this.real).__int__().__eq__(this.real))).value) {
-                buffer.append(((org.python.types.Str) this.real.__int__().__repr__()).value);
-            } else {
-                buffer.append(((org.python.types.Str) this.real.__repr__()).value);
-            }
-        } else {
-            real_present = false;
-        }
-        if (this.real.value != 0 && this.imag.value >= 0) {
-            buffer.append("+");
-        }
-        if (((org.python.types.Bool) ((this.imag).__int__().__eq__(this.imag))).value) {
-            buffer.append(((org.python.types.Str) (this.imag).__int__().__repr__()).value);
+        if (this.real.value != 0.0 || this.real.isNegativeZero()) {
+            return new org.python.types.Str("(" + partToStr(this.real) + ((this.imag.value >= 0.0 && !this.imag.isNegativeZero()) ? "+" : "-") + partToStr(new org.python.types.Float(Math.abs(this.imag.value))) + "j)");
```

Comment:
I'm surprised this wasn't picked up by linting... but we don't use ternary operators, and the line length is *extremely* excessive.

Reply:
Actually we are using ternary at many places in voc, so should we remove all of those ? I'll make that last line bit more readable.

Reply:
We shouldn't be - and the linter *should* be catching them. If it isn't, then we have another problem.